src/NzbDrone.Core/Notifications/Join/JoinProxy.cs

```diff
@@ -50,46 +50,77 @@ public ValidationFailure Test(JoinSettings settings)
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, "Unable to send test Join message. Unknown error.");
+                return new ValidationFailure("ApiKey", ex.Message);
             }
         }
 
         private void SendNotification(string title, string message, RestRequest request, JoinSettings settings)
         {
-            try
-            {
-                var client = RestClientFactory.BuildClient(URL);
 
+            var client = RestClientFactory.BuildClient(URL);
+
+            if (!string.IsNullOrEmpty(settings.DeviceIds))
+            {
+                request.AddParameter("deviceIds", settings.DeviceIds);
+            }
+            else
+            {
                 request.AddParameter("deviceId", "group.all");
-                request.AddParameter("apikey", settings.APIKey);
-                request.AddParameter("title", title);
-                request.AddParameter("text", message);
-                request.AddParameter("icon", "https://cdn.rawgit.com/Sonarr/Sonarr/develop/Logo/256.png"); // Use the Sonarr logo.
+            }
+                
+            request.AddParameter("apikey", settings.ApiKey);
+            request.AddParameter("title", title);
+            request.AddParameter("text", message);
+            request.AddParameter("icon", "https://cdn.rawgit.com/Sonarr/Sonarr/develop/Logo/256.png"); // Use the Sonarr logo.
+
+            var response = client.ExecuteAndValidate(request);
+            var res = Json.Deserialize<JoinResponseModel>(response.Content);
 
-                var response = client.ExecuteAndValidate(request);
-                var res = Json.Deserialize<JoinResponseModel>(response.Content);
+            if (res.success) return;
 
-                if (res.success) return;
+            if (res.userAuthError)
+            {
+                throw new JoinAuthException("Authentication failed.");
+            }
 
-                if (res.errorMessage != null)
+            if (res.errorMessage != null)
+            {
+                // I dislike hard coding this string here but nothing else in the Join API lets you determine the error.
+                // No enum or flag contained in the response for this so we have to do a string check..
+                if(res.errorMessage.Equals("No devices to send to"))
                 {
-                    throw new JoinException(res.errorMessage);
+                    throw new JoinInvalidDeviceException(res.errorMessage);
                 }
-
-                if (res.userAuthError)
+                // Oddly enough, rather than give us an "Invalid API key", the Join API seems to assume the key is valid
+                // But fail on doing a device lookup associated with that key.  
+                // In our case we are using "deviceIds" rather than "deviceId" so when the singular form error shows up
+                // we know our API key was the fault.
```

Comment:
```
// Oddly enough, rather than give us an "Invalid API key", the Join API seems to assume the key is valid,
// but fails when doing a device lookup associated with that key.  
// In our case we are using "deviceIds" rather than "deviceId" so when the singular form error shows up
// we know the API key was the fault.
```